src/gpu/nvidia/sycl_cuda_compat.hpp

```diff
@@ -42,11 +42,23 @@ void host_task(::sycl::handler &cgh, const T &task) {
 
 template <typename native_object_t, typename sycl_object_t>
 native_object_t get_native(const sycl_object_t &sycl_object) {
+    static_assert(!std::is_same_v<native_object_t, CUcontext>,
+            "Use compat::get_native_context() to obtain the native cuda "
+            "context");
     auto handle
             = ::sycl::get_native<::sycl::backend::ext_oneapi_cuda>(sycl_object);
     return reinterpret_cast<native_object_t>(handle);
 }
 
+inline CUcontext get_native_context(const ::sycl::device &sycl_device) {
```

Comment:
Can you please share details on why the standard `sycl::context::get_native()` function was deprecated?